### PR TITLE
added express rate limit to server.js

### PR DESCRIPTION
**Pull request by wilhsie**
hi @tylermaran here's my PR.  I tried testing the change by spamming the graphql endpoint, but I wasn't actually able to get it to error out.  Any ideas why?  I'm going to keep looking into it, maybe I implemented the sample code from the express rate limit docs incorrectly .. all the best

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -44,6 +45,14 @@ app.use(
   })),
 );
 
+// Set express rate limiter
```
Comment (tylermaran) [inline server/index.js:48]: You have this too late in the application. It needs to occur before line 29. 

The rate limiter is a middleware. So any requests to the backend will pass through it. 
So we want to put this in front of the actual /graphql calls.

**Review by tylermaran — CHANGES_REQUESTED** [review]

**Comment by tylermaran** [comment]
Also update this PR to the `rysolv:dev` branch instead of the`master` branch.

File: server/index.js (hunk 1 above)
Comment (wilhsie) [reply server/index.js]: Oh that makes sense.  Thanks!

**Comment by wilhsie** [comment]
For some reason I can't see or select the rysolv:dev branch

Screen Shot 2021-07-18 at 9 03 28 PM

am I missing something?

**Comment by annapo23** [comment]
@wilhsie My bad. Looks like it was temporarily deleted, but it should me there now.

**Comment by wilhsie** [comment]
thanks @annapo23!  I think this should be all set now, but let me know if I missed anything.

File: package.json [hunk 2]
```diff
@@ -91,6 +91,7 @@
     "dotenv": "8.2.0",
     "express": "4.16.4",
     "express-graphql": "0.9.0",
+    "express-rate-limit": "^5.3.0",
```
Comment (annapo23) [inline package.json:94]: Let's remove the caret (`^`), so it should be `"express-rate-limit": "5.3.0",`. That way there are no version differences

File: server/index.js [hunk 3]
```diff
@@ -25,6 +26,14 @@ app.use(cookieParser());
 app.use(express.json({ limit: '10mb' }));
 app.use(express.urlencoded({ limit: '10mb' }));
 
+// Set express rate limiter
+const limiter = rateLimit({
+  windowMs: 1 * 60 * 1000, // 1 minute
+  max: 25, // limit each IP to 25 requests per windowMs
```
Comment (annapo23) [inline server/index.js:32]: Let's alphabetize here.

```
const limiter = rateLimit({
  max: 25, // limit each IP to 25 requests per windowMs
  windowMs: 1 * 60 * 1000, // 1 minute
});
```

Comment (wilhsie) [reply server/index.js:32]: gotcha, no problem

**Comment by wilhsie** [comment]
@annapo23 so I'm getting a deprecation error when trying to commit the two changes your mentioned .. it looks like this:

`(node:7902) [DEP0128] DeprecationWarning: Invalid 'main' field in '/Users/xxxxxx/Documents/code/opensrc/rysolv/node_modules/find-parent-dir/package.json' of 'find-parent-dir.js'. Please either fix that or report it to the module author`

I think this has to do with updating the find-parent-dir package version .. but I'm not 100% sure.  Any ideas?

**Review by tylermaran — APPROVED** [review]

**Comment by tylermaran** [comment]
Hey @wilhsie! Sorry for the delay here. Everything looks good and I tested it out. 

I just bumped up the rate limit to 50 since I noticed the onboarding took up close to the 25 limit. So I was getting the error right after signing up.

**Pull request merged**